### PR TITLE
Added CMClientSettingsComputerRestart

### PR DESCRIPTION
**Pull request by nyanhp**
#### Pull Request (PR) description
Adding new resource CMClientSettingsComputerRestart to configure restart settings.

#### This Pull Request (PR) fixes the following issues
None

#### Task list

- [x] Added an entry to the change log under the Unreleased section of the
      file CHANGELOG.md. Entry should say what was changed and how that
      affects users (if applicable), and reference the issue being resolved
      (if applicable).
- [x] Resource documentation added/updated in README.md.
- [x] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [x] Comment-based help added/updated.
- [x] Localization strings added/updated in all localization files as appropriate.
- [x] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

- - -
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/configmgrcbdsc/107)

### CODE REVIEW

**Comment by nyanhp** [comment]
The failing unit tests seem unrelated to my changes. I am unclear how to proceed.

File: tests/Unit/CMClientSettingsComputerRestart.tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+            }
+
+            Context 'When retrieving Client Policy Settings for Computer Restart' {
+
+                It 'Should return desired results when client settings exist' {
+                    Mock -CommandName Get-CMClientSetting -MockWith { $clientType }
+                    Mock -CommandName Get-CMClientSetting -MockWith { $clientReturn } -ParameterFilter { $Setting -eq 'ComputerRestart' }
+
+                    $result = Get-TargetResource @getInput
+                    $result                                           | Should -BeOfType System.Collections.HashTable
```
Comment (NEllis280) [inline tests/Unit/CMClientSettingsComputerRestart.tests.ps1:70]: Spacing before the pipe char lines 70-78.

File: CHANGELOG.md [hunk 2]
```diff
@@ -39,6 +39,7 @@ version of DSCResource.Common
 - Added CMClientSettingsStateMessaging resource
 - Added CMClientSettingsUserDeviceAffinity resource
 - Added CMSiteConfiguration resource
+- Added CMClientSettingsComputerRestart resource
```
Comment (NEllis280) [inline CHANGELOG.md:42]: Please move to unreleased section.

File: README.md [hunk 3]
```diff
@@ -1685,6 +1687,29 @@ you are using apply and auto correct.
 
 - [CMClientSettingsComputerAgent](Source\Examples\Resources\CMClientSettingsComputerAgent\CMClientSettingsComputerAgent.ps1)
 
+### CMClientSettingsComputerRestart
```
Comment (NEllis280) [inline README.md:1690]: Please add the new resource to supported resources under the reverse dsc section

File: tests/Unit/CMClientSettingsComputerRestart.tests.ps1 (hunk 1 above)
Comment (nyanhp) [reply tests/Unit/CMClientSettingsComputerRestart.tests.ps1]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (nyanhp) [reply CHANGELOG.md]: Done

File: README.md (hunk 3 above)
Comment (nyanhp) [reply README.md:1690]: Done

**Review by NEllis280 — APPROVED** [review]
All comments addressed

**Pull request merged**